src/lib/util/graceful-shutdown.ts

```diff
@@ -1,7 +1,60 @@
-import { Logger } from '../logger';
+import { Logger, LogProvider } from '../logger';
 import { IUnleash } from '../types/core';
+import { GenericUnleashError } from '../error/unleash-error';
 
-function registerGracefulShutdown(unleash: IUnleash, logger: Logger): void {
+export interface IGracefulShutdown {
```

Comment:
This is not an interface for the shutdown itself but for a manager of shutdowns.
```suggestion
export interface IGracefulShutdownHookManager {
```

Reply:
hmm, I'm not sure if we want the manager in the interface name. What do you think of `IGracefulShutdownHookRegistry` ? I think I like that better for the class as well.

Reply:
Fixed in cb54915dd